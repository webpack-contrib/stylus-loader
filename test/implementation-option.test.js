/**
 * @jest-environment node
 */

const {
  compile,
  getCodeFromBundle,
  getCodeFromStylus,
  getCompiler,
  getErrors,
  getWarnings,
} = require("./helpers");

jest.setTimeout(30000);

describe("implementation option", () => {
  it("should work", async () => {
    const testId = "./basic.styl";
    const compiler = getCompiler(testId, {
      // eslint-disable-next-line global-require
      implementation: require("stylus"),
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromStylus = await getCodeFromStylus(testId);

    expect(codeFromBundle.css).toBe(codeFromStylus.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should work when implementation option is string", async () => {
    const testId = "./basic.styl";
    const compiler = getCompiler(testId, {
      implementation: require.resolve("stylus"),
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);
    const codeFromStylus = await getCodeFromStylus(testId);

    expect(codeFromBundle.css).toBe(codeFromStylus.css);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should throw error when unresolved package", async () => {
    const testId = "./basic.styl";
    const compiler = getCompiler(testId, {
      implementation: "unresolved",
    });
    const stats = await compile(compiler);

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
