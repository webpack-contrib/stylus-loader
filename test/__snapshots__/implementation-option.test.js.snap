// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`implementation option should throw error when unresolved package: errors 1`] = `
[
  "ModuleBuildError: Module build failed (from \`replaced original path\`):
NonErrorEmittedError: (Emitted value instead of an instance of Error) Error: Cannot find module 'unresolved' from 'src/utils.js'",
]
`;

exports[`implementation option should throw error when unresolved package: warnings 1`] = `[]`;

exports[`implementation option should work when implementation option is string: css 1`] = `
"body {
  font: 12px Helvetica, Arial, sans-serif;
}
a.button {
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  border-radius: 5px;
}
"
`;

exports[`implementation option should work when implementation option is string: errors 1`] = `[]`;

exports[`implementation option should work when implementation option is string: warnings 1`] = `[]`;

exports[`implementation option should work: css 1`] = `
"body {
  font: 12px Helvetica, Arial, sans-serif;
}
a.button {
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  border-radius: 5px;
}
"
`;

exports[`implementation option should work: errors 1`] = `[]`;

exports[`implementation option should work: warnings 1`] = `[]`;
